src/main/java/com/lists/LinkedList.java

```diff
@@ -0,0 +1,192 @@
+/*
+ * File: LinkedList.java
+ * Author: Tarun Jain
+ * Date: 1 Feb, 2016
+ * -----------------------------------------
+ * LinkedList is a generic data structure to store data items
+ * of any type in list fashion.  
+ * 
+ *  
+ */
+package com.lists;
+
+import java.util.Iterator;
+import java.util.NoSuchElementException;
+
+/**
+ * LinkedList is a generic data structure to store elements in 
+ * list manner.
+ * 
+ * LinkedList provides following API
+ * ----------------------------------------------
+ *         LinkedList ()                   // constructs a empty list.
+ * int     size()                      // returns the length of list in integer.
+ * boolean isEmpty()               // returns true if list is empty, false otherwise.
+ * void    add(Item item)      	   // append an item in the last of list.
+ * void    add(Item item, int index)  //append an item at the given index in the list.
+ * Item    get(int index)             // returns the item at the given index from the list.
+ * boolean remove(int index)	   // remove the node and return true at the given index from the list.
+ * Iterator<Item> iterator()   	   // Iterate over the linkedList
+ *
+ * @author Tarun Jain
+ * @version 1.0
+ * @param <Item>
+ * 
+ */
+
+public class LinkedList<Item> {
```

Comment:
Also, every `Iterable` data structure must implement ``Iterable<item>` interface. Then and only, the for each loop and other code that expects an iterable DS can implicitly detects that it is iterable. Correct it before merging.